File: p3and4/practico3and4/api/serializers/compra.py
from rest_framework import serializers
from practico3and4.models.compra import Compra, DetalleCompra
from .libro import LibroSerializer

class DetalleCompraSerializer(serializers.ModelSerializer):
    libro = LibroSerializer(read_only=True)

    class Meta:
        model = DetalleCompra
        fields = ['id', 'libro', 'precio_unitario']

class CompraSerializer(serializers.ModelSerializer):
    detalles = DetalleCompraSerializer(many=True, read_only=True)
    comprobante_pago = serializers.ImageField(required=False)

    class Meta:
        model = Compra
        fields = ['id', 'usuario', 'fecha', 'total', 'qr_generado', 'comprobante_pago', 'confirmada', 'detalles']
        read_only_fields = ['usuario', 'fecha', 'qr_generado', 'confirmada']

File: LibreriaFront/src/types/Compra.ts
export interface DetalleCompra {
  id: number;
  precio_unitario: number;
  libro: {
    id: number;
    titulo: string;
    autor: string;
  };
}

export interface Compra {
  id: number;
  fecha: string;
  total: number | string;
  qr_generado: string;
  comprobante_pago?: string;
  confirmada: boolean;
  detalles: DetalleCompra[];
  usuario: number;
}
